From f641c9a5d4820c1ac6783e393e99e7d5128fdac0 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Mon, 16 Apr 2018 01:23:27 +0300
Subject: [PATCH 5/5] recovery: Fix loading time from /persist

* We have added time_off_3_data but never used it

Change-Id: I334d7f5a2af6168227a84f551d618f9f661d695a
---
 recovery.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/recovery.cpp b/recovery.cpp
index e90d28de..3db29ed7 100644
--- a/recovery.cpp
+++ b/recovery.cpp
@@ -1502,13 +1502,19 @@ static void copy_userdata_files() {
         if (access(time_off_root, F_OK) != 0) {
             if (access(time_off_1_data, R_OK) == 0) {
                 file_copy(time_off_1_data, time_off_root);
-            }
-            else if (access(time_off_2_data, R_OK) == 0) {
+            } else if (access(time_off_2_data, R_OK) == 0) {
                 file_copy(time_off_2_data, time_off_root);
             }
         }
         ensure_path_unmounted("/data");
     }
+
+    if (ensure_path_mounted("/persist") == 0) {
+        if (access(time_off_3_data, R_OK) == 0) {
+            file_copy(time_off_3_data, time_off_root);
+        }
+        ensure_path_unmounted("/persist");
+    }
 }
 
 static void set_time() {
-- 
2.17.0

